infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
> What makes you believe that we can always factor out a monadic-context-free MyData piece?

It was an examples, it's hard to talk about without knowing which interface may be exposed by time-warp.

> Its internal workings depend upon the application monad: we can't pull out a simple MyData from such a relayer. It really truly does depend upon the ultimate application monad.

Again, I see no problem. I wrote this:
```
myRelay :: MonadMockable m => MyData -> RelayFunctions m
```
Here you are free to change `MonadMockable m` to whatever constraints you need. You can add `MonadIO` or `MonadReader Something` if you want.

`rcfgDelay :: NodeId -> m (Maybe Microsecond)` can be put into `RelayFunctions` as well as `rcfgNextSlotLeader :: m (Maybe NodeId)`. Perhaps you already have the type with these functions, I just don't know its name, so I call it `RelayFunctions`.
Also you definitely should have some way to create these `RelayFunctions`. It is most likely a function which takes some data (or it can take nothing). I called it `myRelay`. It returns `RelayFunctions m` and you can requrest whatever constraints you need on `m`.